src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -532,6 +561,15 @@ private static string GenerateUpgradeReport(string projectName, string backupPat
                         updatedPackagesNew,
                         frameworks);
 
+                    if (userAction != null && !userAction.IsSolutionLevel && userAction.Action == NuGetProjectActionType.Install && uiService.Projects?.Count() == 1)
```

Comment:
offline discuss with @jebriede : add variable to save pre-computed IsPackageTransitive condition, and set in telemetry if pre-computed variable is set.